ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp

```diff
@@ -149,19 +149,22 @@ double window_get_alpha() {
 
 void window_set_position(int x, int y)
 {
+  if (window_get_fullscreen()) return;
   enigma::windowX = x;
   enigma::windowY = y;
   SetWindowPos(enigma::hWnd, HWND_TOP, enigma::windowX, enigma::windowY, 0, 0, SWP_NOSIZE|SWP_NOACTIVATE);
 }
 
 void window_set_size(unsigned int width, unsigned int height)
 {
+  if (window_get_fullscreen()) return;
   enigma::windowWidth = width;
   enigma::windowHeight = height;
   enigma::compute_window_size();
 }
 
 void window_set_rectangle(int x, int y, int width, int height) {
+  if (window_get_fullscreen()) return;
```

Comment:
None of this is correct, I tested these in GMS and GM8 at one point. The size that you set will be applied if the window is changed back to not fullscreen. The request can't be simply ignored.

Reply:
This i did as a temporary fix because fullscreen breaks when you use these functions. The window will be be switched to borderless windowed mode rather than stay fullscreen, when these are called. No one should be attempting to resize the window unless they are in windowed mode, anyway.

Reply:
but, what i can do, is save the values to windowX, windowY, windowWidth, and windowHeight, so that it will do exactly what you want still, now that i tink about it. Sound good?

Reply:
@RobertBColton i tried removing the `if (window_get_fullscreen()) return;` and no matter what i do, attempting to use those functions to resize or reposition the window break both xlib and windows when used in fullscreen,

Reply:
@RobertBColton are you going to address what I said here or are you just going to ignore it because you don't know how to create a better workaround for this than me.

Reply:
Allowing these functions to work when fullscreen produces glitchy behavior on both windows and xlib. What do you want me to do? Are you just going to refuse to merge this because you prefer glitchy behavior over more correct behavior?

Reply:
What GM obviously does, is save the window x/y/w/h to global state variables, and whatever these are set to, the game will go to when switching out of fullscreen. If we did excessive state checks like this in our main game loop like GM does, ENIGMA would suffer major performance with wasteful if statements called every step for every single last cached window state, among many other things GM does this with.

I thought you were against global state variables, anyway?

Reply:
Is there any way this could be addressed in our follow up pr?